File: JAVA/src/_Searching/BinarySearch.java
package _Searching;

public class BinarySearch {

    public static int getSearchResult(int arr[], int size , int key){

        int s=0;
        int e=size-1,mid;

        while (s<=e)
        {
            //mid=(s+e)/2;
            mid=s+(e-s)/2;
            if (arr[mid]==key)
            {
                return mid;
            }
            else if (arr[mid]>key)
                e=mid-1;
            else
                s=mid+1;
        }


        return -1;
    }
    public static void main(String[] args) {
        int arr[] = {1,2,3,4,5};  // arr must be sorted in binary search
        int size =5;
        int key=6;
        int res = getSearchResult(arr,size,key);

        if (res==-1)
        {
            System.out.println("Element not found");
        }
        else
            System.out.println("Element found at "+ res + " index");
    }
}

File: CPP/Array/Seaching/BinarySearch.cpp
#include <iostream>
using namespace std ;

int binarySearch(int arr[],int size, int element)
{
    int start=0 , end = size-1 ,mid ;
    while(start<=end)
    {
         mid = (start+end)/2;
         if (arr[mid]==element)
         return mid;
         else if (arr[mid]>element)
         end = mid -1;
         else
         start=mid+1;
    } 
    return -1;
}

int recursiveBinarySearch(int arr[], int start , int end , int element)
{
    if (start<=end)
    {
        int mid = (start+end)/2 ;
        if (arr[mid]==element)
        {
            return mid;
        }
        else if (arr[mid]>element)
        return recursiveBinarySearch(arr,start,mid-1,element);

        else
        return recursiveBinarySearch(arr,mid+1,end,element);
    }
    return -1;
}
int main(){
    int size,element;
    cin>> size; 
    int arr[size];
    for (int i =0 ;i<size;i++)
    cin>>arr[i];

    cin>>element;

   if (binarySearch(arr,size,element)==-1)
    cout<<"element not found"<<endl;
   else
   cout<<"element found at "<<binarySearch(arr,size,element)<<" location"<<endl;
   
   cout<<"by using recursive binary search "<<recursiveBinarySearch(arr,0,size-1,element)<<endl;
}